presto-main/src/main/java/com/facebook/presto/sql/planner/plan/Assignments.java

```diff
@@ -0,0 +1,188 @@
+import com.google.common.collect.Maps;
+
+import java.util.Collection;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.stream.Collector;
+import java.util.stream.StreamSupport;
+
+import static com.facebook.presto.util.ImmutableCollectors.toImmutableMap;
+import static com.google.common.base.Preconditions.checkState;
+import static java.util.Objects.requireNonNull;
+
+public class Assignments
+{
+    public static Builder builder()
+    {
+        return new Builder();
+    }
+
+    public static Assignments identity(Iterable<Symbol> symbols)
+    {
+        return builder().putIdentities(symbols).build();
+    }
+
+    public static Assignments of()
+    {
+        return builder().build();
+    }
+
+    public static Assignments of(Symbol symbol, Expression expression)
+    {
+        return builder().put(symbol, expression).build();
+    }
+
+    public static Assignments of(Symbol symbol1, Expression expression1, Symbol symbol2, Expression expression2)
+    {
+        return builder().put(symbol1, expression1).put(symbol2, expression2).build();
+    }
+
+    private final Map<Symbol, Expression> assignments;
+    private final List<Symbol> outputs;
+
+    @JsonCreator
+    public Assignments(@JsonProperty("assignments") Map<Symbol, Expression> assignments)
+    {
+        this.assignments = ImmutableMap.copyOf(requireNonNull(assignments, "assignments is null"));
+        this.outputs = ImmutableList.copyOf(assignments.keySet());
+    }
+
+    public List<Symbol> getOutputs()
+    {
+        return outputs;
+    }
+
+    public Map<Symbol, Expression> getAssignments()
+    {
+        return assignments;
+    }
+
+    public <C> Assignments rewrite(ExpressionRewriter<C> rewriter)
+    {
+        return rewrite(expression -> ExpressionTreeRewriter.rewriteWith(rewriter, expression));
+    }
+
+    public Assignments rewrite(Function<Expression, Expression> rewrite)
+    {
+        return assignments.entrySet().stream()
+                .map(entry -> Maps.immutableEntry(entry.getKey(), rewrite.apply(entry.getValue())))
+                .collect(toAssignments());
+    }
+
+    public Assignments filter(Collection<Symbol> symbols)
+    {
+        return assignments.entrySet().stream()
+                .filter(entry -> symbols.contains(entry.getKey()))
+                .collect(toAssignments());
+    }
+
+    private Collector<Entry<Symbol, Expression>, Builder, Assignments> toAssignments()
+    {
+        return Collector.of(
+                Assignments::builder,
+                (builder, entry) -> builder.put(entry.getKey(), entry.getValue()),
+                (left, right) -> {
+                    left.putAll(right.build());
+                    return left;
+                },
+                Assignments.Builder::build);
+    }
+
+    public Assignments append(Symbol symbol, Expression expression)
```

Comment:
maybe remove this method? There is just one usage with can be rewritten to using builder. I think builder is convention in presto.